nautobot/core/ui/object_detail.py

```diff
@@ -0,0 +1,598 @@
+
+    def render_content(self, request, instance):
+        if self.template_string is not None:
+            return Template(self.template_string).render(RequestContext(request, {"object": instance}))
+        return render(request, self.template_path, {"object": instance})
+
+
+class ObjectFieldsPanel(Panel):
+    """A panel that renders a table of object instance attributes."""
+
+    def __init__(self, *, fields="__all__", field_transforms=None, **kwargs):
+        """
+        Instantiate an ObjectFieldsPanel.
+
+        Args:
+            fields (str, list): The string "__all__", or an ordered list of field names to display.
+            field_transforms (dict): A dict of `{field_name: [transform_functions]}` that can be used to customize the
+                                     display string for any given field.
+        """
+        self.fields = fields
+        self.field_transforms = field_transforms or {}
+        super().__init__(**kwargs)
+
+    def render(self, request, instance):
+        """Override Panel.render() to render an attr-table as the panel body."""
+        return format_html(
+            """\
+<div class="panel panel-default">
+    <div class="panel-heading">
+        <strong>{label}</strong>
+    </div>
+    <table class="table table-hover panel-body attr-table">
+        {content}
+    </table>
+</div>""",
+            label=self.render_label(request, instance),
+            content=self.render_content(request, instance),
+        )
+
+    def render_label(self, request, instance):
+        if self.label is None:
+            return bettertitle(instance._meta.verbose_name)
+        return super().render_label(request, instance)
+
+    def render_content(self, request, instance):
+        """Render the table rows corresponding to the specified fields, applying field_transforms as specified."""
+        result = format_html("")
+        fields = self.fields
+
+        if fields == "__all__":
+            # Derive the list of fields from the instance, skipping certain fields by default.
+            fields = []
+            for field in instance._meta.get_fields():
+                if field.hidden or field.name.startswith("_"):
+                    continue
+                if field.name in ("id", "created", "last_updated"):
+                    # Handled elsewhere in the detail view
+                    continue
+                if field.is_relation and (field.many_to_many or field.one_to_many):
+                    continue
+                fields.append(field.name)
+
+        for field_name in fields:
+            field = instance._meta.get_field(field_name)
+            field_label = bettertitle(field.verbose_name)
+            field_value = getattr(instance, field.name)
+
+            if field_name in self.field_transforms:
+                field_display = field_value
+                for transform in self.field_transforms[field_name]:
+                    field_display = transform(field_display)
```

Comment:
~What would be a case where we have multiple field transforms for a single field?~

Reply:
Nevermind I just have to look👀